Otus-DevOps-2018-02/ifqthenp_microservices
# Homework 14. Docker containers (docker-2)

# Completed Homework №14

:large_blue_diamond: [x] Main tasks
:large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

- :large_blue_diamond: Created docker-host on GCP instance using `docker-machine` command
- :large_blue_diamond: Created Dockerfile with configuration necessary to build `reddit` image
- :large_blue_diamond: Pushed newly built Docker `reddit` image to Docker Hub as `ifqthenp/otus-reddit:1.0`
- :large_orange_diamond: Created Packer template `docker.json` to build an image with preinstalled Docker CE
- :large_orange_diamond: Created Terraform configuration allowing to create several `app` instances (2 by default)
- :large_orange_diamond: Created Ansible playbooks to deploy `ifqthenp/otus-reddit:1.0` application from Docker Hub to GCP instances

## How to get project running

Run command `docker run --name reddit -d -p 9292:9292 ifqthenp/otus-reddit:1.0`

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Run command `curl -I localhost:9292`

## PR checklist
